docs/source/tools/trigger-service.rst

```diff
@@ -0,0 +1,119 @@
+
+Although as we'll see, the trigger-service exposes an endpoint for end-users to upload DAR files to the service (DAML archives containing triggers), it is sometimes convenient to start the service preconfigured with a specific DAR. To do this, the ``--dar`` option is provided.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key --dar <DAR>
+
+Identifying trigger parties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we briefly explain how parties are identified in end-user interactions with a running trigger service.
+
+When an end-user interacts with a trigger-service, they do so by sending HTTP requests to one of several trigger-service HTTP endpoints. Such requests must contain information detailing the party the trigger-service is to act on the behalf of in its handling of the user request.
+
+This party information is conveyed via  HTTP basic authentication headers where the requesting party is represented by a unique username/password pair (Later versions of this documentation will explain running the trigger-service in authenticated contexts. For now, we assume non-authenticated contexts. In these contexts, the passwords associated with parties are not important and may have any values at all but they still must be provided).
+
+With an understanding of the role of credentials now in mind we can explain the ``--no-secret-key`` parameter passed to the trigger-service in the preceding section. When the trigger-service stores credentials it first encrypts them. The encryption/decryption algorithms it uses to do so require the use of a "key". What key is to be used is expected to be provided as the value of the environment variable ``TRIGGER_SERVICE_SECRET_KEY`` (for example, before starting the trigger-service by a command like ``export TRIGGER_SERVICE_SECRET_KEY="my secret key"``). At startup, if the trigger-service detects that the environment variable is not defined then it will terminate unless the parameter ``--no-secret-key`` is in its startup options. If it is, the service will continue with a default key.
+
+If the party 'alice' has password '&alC2l3SDS*V' then we can construct her credentials token by joining her username to her password separated by ':' giving 'alice:&alC2l3SDS*V'. Then base64 encoding her credentials token gives 'YWxpY2U6JmFsQzJsM1NEUypW'. We will see example usage of the credential tokens produced in this fashion in the next section where examples are given sending requests to the trigger-service.
+
+End-user interaction with the trigger-service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``health``
```

Comment:
Given that these requests are not sorted alphabetically, I would suggest to sort them by importance. The first thing I want to know is how I can start a trigger, then how I can stop a trigger. Then probably listing and then it starts to become less important in general so the order matters less as well. The health check is probably the very last thing anyone cares about when they are just starting out.

Reply:
there's a bit of a narrative there though - upload dar - start trigger -stop trigger - list - status - health. how do you feel about that?

Reply:
I would still move uploading later since you can skip it if you pass in the DAR. Apart from that sounds like a great order :+1:

Reply:
✔️